content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md

```diff
@@ -0,0 +1,197 @@
+storage; this change isn't relevant to storage drivers that are built in to Kubernetes.
+
+To turn on this new, alpha feature, you enable the `CSINodeExpandSecret` feature
+gate for the kube-apiserver and kubelet, which turns on a mechanism to send `secretRef`
+configuration as part of NodeExpansion  by the CSI drivers thus make use of
+the same to perform node side expansion operation with the underlying
+storage system.
+
+## What is this all about?
+
+Before Kubernetes v1.24, you were able to define a cluster-level StorageClass
+that made use of [StorageClass Secrets](https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html),
+but you didn't have any mechanism to specify the credentials that would be used for
+operations that take place when the storage was mounted onto a node and when
+when the volume has to be expanded at node side.
+
+The Kubernetes CSI already implemented a similar mechanism specific kinds of
+volume resizes; namely, resizes of PersistentVolumes where the resizes take place
+independently from any node. In that case, you associate a PersistentVolume
+with a Secret that contains credentials for volume resize actions.
+CSI also supports a `nodeExpandVolume` operation, where the resize is driven
+from a node in your cluster where the volume is attached. Having resize (expansion)
+operations sometimes able to use credentials and sometimes not meant it was difficult
+for people to design CSI drivers that could cover all resize scenarios.
+
+- At times, the CSI driver needs to check the actual size of the backend block storage (or image)
+  before proceeding with a node-level filesystem expand operation. This avoids false positive returns
+  from the backend storage cluster during filesystem expands.
+
+- When a PersistentVolume represents encrypted block storage (for example using LUKS)
+  you need to provide a passphrase in order to expand the device, and also to make it possible
+  to grow the filesystem on that device.
+
+- For various validations at time of node expansion, the CSI driver has to be connected
+  to the backend storage cluster. If the` nodeExpandVolume` request includes a `secretRef`
+  then the CSI driver can make use of the same and connect to the storage cluster to
+  perform the cluster operations.
+
+## How does it work?
+
+To enable this functionality from this version of Kubernetes, Kubernetes Storage SIG have introduced
+a new feature gate called `CSINodeExpandSecret`. Once the feature gate is enabled
+in the cluster, NodeExpandVolume requests can include a `secretRef` field. The NodeExpandVolume request
+is part of CSI; for example, in a request which has been sent from the Kubernetes
+control plane to the CSI driver.
+The admin can specify these secrets as an opaque parameter in storage class as
+they specifies other CSI secrets today.  The secrets mention in the storage class
+has to carry the key name as shown below
```

Comment:
Is this right?
```suggestion
As a cluster operator, you admin can specify these secrets as an opaque parameter in a StorageClass,
the same way that you can already specify other CSI secret data. The StorageClass needs to have some
CSI-specific parameters set. Here's an example of those parameters:
```

(these look like annotations or labels, not fields)

Reply:
Yeah.. these are parameters in SC.. but the format is like annotations though.